ls1intum/Artemis
# Show exercise participations for course on Team page

### Checklist

### Motivation and Context
It would be nice if the Team page offered some information about a team's activity in the course so that it is easier to keep track of a team's progression for tutors and instructors.

Since teams exist on an exercise-level and not on a course-level, teams with the same short name in the same course are regarded as the "same" team.

_Example of a workflow: A course has five exercises, all of which should be worked on in the same team constellation. Tutors create teams together with the students for the first exercise and the students get started. For subsequent exercises, the instructor uses the "Import teams" feature to clone the existing teams of the first exercise into the second exercise. Cloning will keep the team short names. We know that teams across exercises that have the same short name correspond to the same real-life team._

The Team page will list all of the participations of a team (technically: of all teams with the same short name) for a course. It shows exercise title and release date for context and then details about the participation (start date, number of submissions and status).

### Description
1. Add REST endpoint `GET /courses/:courseId/teams/:teamShortName/with-exercises-and-participations` (get course with all released exercises, for which a team exists by that short name, and their participations for the team)
2. Add component `TeamParticipationTable` that displays the exercises in the course (for which the team exists) and the participation details for the team
3. Include this new component on the Team page and let it be visible for students who belong the the team and for TAs and above (instructors can also look into the submissions by clicking on them, as it is done on the regular participation page; admins see additional id columns useful for debugging)

### Steps for Testing
1. Log in to Artemis as a TA or above
2. Navigate to Course Management
3. Go into a course that has multiple Team Exercises
4. Click on a button "Team" for one of those exercises
5. Verify that the team page shows all exercises for this team short name in the course with participation details

### Screenshots
![OnPaste 20200429-170009](https://user-images.githubusercontent.com/7109225/80611548-06cca580-8a3b-11ea-9ba6-5369067e5d04.png)
**Left:** View as admin (in German), **Right:** View as student (or non-team-owning tutor) (in English)

![OnPaste 20200429-171116](https://user-images.githubusercontent.com/7109225/80612590-72fbd900-8a3c-11ea-92ba-8148d5885dae.png)
**Above:** View as instructor
